src/SparkFun_AS7331.h

```diff
@@ -14,1312 +14,716 @@
+
+    /// @brief Performs a soft reset of the sensor.
+    /// @return True if successful, false otherwise.
+    bool reset(void);
+
+    /// @brief Reads the sensor's temperature, converts it to a usable form, and
+    /// saves it to the internal temperature variable.
+    /// @return 0 if successful, negative if error, positive for warning.
+    sfeTkError_t readTemp(void);
+
+    /// @brief Reads the sensor's UVA register, converts it to a usable form, and
+    /// saves it to the internal UVA variable.
+    /// @return 0 if successful, negative if error, positive for warning.
+    sfeTkError_t readUVA(void);
+
+    /// @brief Reads the sensor's UVB register, converts it to a usable form, and
+    /// saves it to the internal UVB variable.
+    /// @return 0 if successful, negative if error, positive for warning.
+    sfeTkError_t readUVB(void);
+
+    /// @brief Reads the sensor's UVC register, converts it to a usable form, and
+    /// saves it to the internal UVC variable.
+    /// @return 0 if successful, negative if error, positive for warning.
+    sfeTkError_t readUVC(void);
+
+    /// @brief Read's all three UV registers, converts them to a usable form, then
+    /// saves them to their respective internal variable.
+    /// @return 0 if successful, negative if error, positive for warning.
+    sfeTkError_t readAllUV(void);
+
+    /// @brief Read the sensor's temperature, UV, and external time conversion
+    /// clock counts, converts them, and then saves them to their respective
+    /// internal variable.
+    /// @return 0 if successful, negative if error, positive for warning.
+    sfeTkError_t readAll(void);
+
+    /// @brief Read the conversion clock counts register and saves it to the
+    /// internal output conversion time variable.
+    /// @return 0 if successful, negative if error, positive for warning.
+    sfeTkError_t readOutConv(void);
+
+    /// @brief Returns the last valid UVA reading.
+    /// @return A float of the UVA reading.
+    float getUVA(void);
+
+    /// @brief Returns the last valid UVB reading.
+    /// @return A float of the UVB reading.
+    float getUVB(void);
+
+    /// @brief Returns the last valid UVC reading.
+    /// @return A float of the UVC reading.
+    float getUVC(void);
+
+    /// @brief Returns the last valid Temperature reading.
+    /// @return A float of the Temperature reading.
+    float getTemp(void);
+
+    /// @brief Returns the last valid output conversion.
+    /// @return The output conversion time in number of clock cycles.
+    uint32_t getOutConv(void);
+
+    /// @brief Getter for the currently configured gain.
+    /// @return Sensor's gain expressed as (1 << (11 - gain)).
+    as7331_gain_t getGain(void);
+
+    /// @brief Sets the UV sensor's gain.
+    /// @param gain The gain to set the sensor to.
+    /// @return 0 if successful, negative if error, positive for warning.
+    sfeTkError_t setGain(const as7331_gain_t &gain);
```

Comment:
Reduces copying of existing data. Constructor could be passed by reference. `begin` does besides the bus. `setCommunicationBus` and `begin` use a pointer because they're setting an internal pointer. References are checked at compile time, and the constructor doesn't define a default Bus to use. `runDefaultSetup` could use a reference, so I'll do that. It should also be const. Same with `prepareMeasurement`.